packages/ragbits-core/src/ragbits/core/llms/litellm.py

```diff
@@ -83,3 +85,24 @@ def count_tokens(self, prompt: BasePrompt) -> int:
             Number of tokens in the prompt.
         """
         return sum(litellm.token_counter(model=self.model_name, text=message["content"]) for message in prompt.chat)
+
+    def _format_chat_for_llm(self, prompt: BasePrompt) -> ChatFormat:
+        images = prompt.list_images()
+        chat = prompt.chat
+        if images:
+            if not litellm.supports_vision(self.model_name):
+                warnings.warn(
+                    message=f"Model {self.model_name} does not support vision. Image input would be ignored",
+                    category=UserWarning,
+                )
+                return chat
+            user_message = chat[-1]
```

Comment:
We cannot safely assume that the last message is the `user_message`. There may be few show examples. You should actually look at the type specified in the message.

Reply:
actually in prompt.chat we always add user message at the end - few shots (if exist) are added before

Reply:
Ok, I still don't feel that blindly taking the last message is reasonable, when they have their types specified explicitly. This is a big assumption that may be correct today, but doesn't have to true moving forward - since the types are specified, let's make use of the fact.

Reply:
ok - so the goal of code below is to add image input to the user message. should we than:

a) search for the last user message in the conversation and add image input to it ?
b) if the last message type is not user ignore it and raise some warning ?

Reply:
btw - regarding few shots i think we should create a backlog ticket to support images in few shots as well

Reply:
> btw - regarding few shots i think we should create a backlog ticket to support images in few shots as well

Definitely - I wanted to suggest the same 😃

Reply:
Another related feature for a separate ticket is allowing `Prompts` with no text user-message in cases when there are images specified (i.e., support for scenario when the prompt consists of system prompt and image-only user prompt)

Reply:
@kdziedzic68 both (a) and (b) are ok with me - so whatever you prefer